tests/Unit/IO/H5/Python/Test_ExtractDatFromH5.py

```diff
@@ -76,8 +80,22 @@ def test_extract_dat_files(self):
         self.assertEqual(memory_data.all(), expected_memory_data.all())
         self.assertEqual(timestep_data.all(), expected_timestep_data.all())
 
+        if os.path.exists(self.created_dir):
+            shutil.rmtree(self.created_dir)
+
+        # Parallelize. Use 2 cores (this is all we get on CI)
+        extract_dat_files(self.h5_filename,
+                          out_dir=self.created_dir,
+                          num_cores=2)
+
+        memory_data = np.loadtxt(memory_data_path)
+        timestep_data = np.loadtxt(timestep_data_path)
+
+        self.assertEqual(memory_data.all(), expected_memory_data.all())
+        self.assertEqual(timestep_data.all(), expected_timestep_data.all())
```

Comment:
I don't think this does what you want. The `.all()` actually evaluates the boolean value of all elements in the array, which is true if all elements are nonzero. To compare the two arrays, `import numpy.testing as npt` and use `npt.assert_almost_equal`. Fix occurrence above too please.